dask_jobqueue/core.py

```diff
@@ -203,12 +203,16 @@ def __init__(
         if shebang is None:
             shebang = dask.config.get("jobqueue.%s.shebang" % config_name)
 
-        if cores is None:
+        if (cores is None) and (memory is None):
             raise ValueError(
                 "You must specify how many cores to use per job like ``cores=8``"
```

Comment:
I would tweak the error messages slightly, e.g. something like (not tested):
```py
"You must specify how much cores and memory per job you want to use, for example:\n"
"cluster = {}(cores={}, memory='{!r}'".format(self.__class__, cores or 8, memory or '24GB')
```

I would also use the same error message in all cases for simplicity reasons